libs/sdk-react-native/android/src/main/java/com/breezsdk/BreezSDKModule.kt

```diff
@@ -1,27 +1,236 @@
+    fun startLogStream(promise: Promise) {
+        try {
+            var emitter = reactApplicationContext
+                    .getJSModule(RCTDeviceEventEmitter::class.java)
+
+            setLogStream(BreezSDKLogStream(emitter))
+            promise.resolve("Log stream started")
+        } catch (e: SdkException) {
+            e.printStackTrace()
+            promise.reject(TAG, "Error calling setLogStream", e)
+        }
+    }
+
+    @ReactMethod
+    fun initServices(apiKey: String, deviceKey: ReadableArray, deviceCert: ReadableArray, seed: ReadableArray, promise: Promise) {
+        if (this.breezServices != null) {
+            promise.reject(TAG, "BreezServices already initialized")
+        }
+
+        var emitter = reactApplicationContext.getJSModule(RCTDeviceEventEmitter::class.java)
+        var creds = GreenlightCredentials(asUByteList(deviceKey), asUByteList(deviceCert))
+        var config = defaultConfig(EnvironmentType.PRODUCTION)
+        config.apiKey = apiKey
 
-            promise.resolve(response);
+        try {
+            this.breezServices = initServices(config, asUByteList(seed), creds, BreezSDKListener(emitter))
+            promise.resolve("BreezServices initialized")
         } catch (e: SdkException) {
-            e.printStackTrace();
-            promise.reject("SdkException");
+            e.printStackTrace()
+            promise.reject(TAG, "Error calling initServices", e)
+        }
+    }
+
+    @ReactMethod
+    fun start(promise: Promise) {
+        this.breezServices?.let {breezServices->
+            try {
+                breezServices.start()
+                promise.resolve("BreezServices started")
+            } catch (e: SdkException) {
+                e.printStackTrace()
+                promise.reject(TAG, "Error calling start", e)
+            }
+        } ?: run {
+            promise.reject(TAG, "BreezServices not initialized")
+        }
+    }
+
+    @ReactMethod
+    fun sync(promise: Promise) {
+        this.breezServices?.let {breezServices->
+            try {
+                breezServices.sync()
+                promise.resolve("BreezServices syncing")
+            } catch (e: SdkException) {
+                e.printStackTrace()
+                promise.reject(TAG, "Error calling sync", e)
+            }
+        } ?: run {
+            promise.reject(TAG, "BreezServices not initialized")
+        }
+    }
+
+    @ReactMethod
+    fun stop(promise: Promise) {
+        this.breezServices?.let {breezServices->
+            try {
+                breezServices.stop()
+                promise.resolve("BreezServices stopped")
+            } catch (e: SdkException) {
+                e.printStackTrace()
+                promise.reject(TAG, "Error calling stop", e)
+            }
+        } ?: run {
+            promise.reject(TAG, "BreezServices not initialized")
+        }
+    }
+
+    @ReactMethod
+    fun sendPayment(bolt11: String, amountSats: String?, promise: Promise) {
```

Comment:
@roeierez I'm testing changing the type to optional Double:
```
    fun sendPayment(bolt11: String, amountSats: Double?, promise: Promise) {
```
but when mirroring it in javascript:
```
export async function sendPayment(bolt11: string, amountSats?: number): Promise<Payment> {
```
it causes an error if no amountSats is passed:
```
Attempt to invoke virtual method double java.lang.Double.doubleValue() on a null object reference
```
Is it possible to send a 0 amountSats in the case no param is provided? Does it affect the sdk? Other options could be to wrap all params over the bridge in an object, or passing as a string.

Reply:
Which component is trying to call `java.lang.Double.doubleValue() `? According to the React Native docs for this case we should use number? in JS (as you just did) and Double in kotlin (not Double?), see here: https://reactnative.dev/docs/native-modules-android#argument-types

Also I admit I don't understand why this might help, worth trying as the docs clearly shows it should be supported.

Reply:
Also how do you pass it from the typescript to the native code? As is without conversion? Maybe if you can paste here the JS code I would be able to help.

Reply:
I tried also with using `number?` in JS and `Double` in kotlin. Same error. Below is the typescript code, calling without any conversion:
```
export async function sendPayment(bolt11: string, amountSats?: number): Promise<Payment> {
    const response = await BreezSDK.sendPayment(bolt11, amountSats)
    return response as Payment
}
```
if the function is called with only a bolt11 param the error occurs
```

await sendPayment("lnbc1p37jj09pp5rt6...")
```

Reply:
Here the stack trace from react native:
<img width="195" alt="Screenshot 2023-02-16 at 10 32 01" src="https://user-images.githubusercontent.com/551697/219325697-0b2330df-a7d4-4472-935d-aa07ecef276f.png">

Reply:
@dangeross I have looked at the bridge code and indeed there is not way to pass nullable primitives.
In that case I think your suggestion to pass 0 from the typescript code to the native code and then again null from the native code to the sdk is the best option.